### ISSUE TITLE
How to display <video> with <img> as fallback?

### ISSUE THREAD

**Issue by binarykitchen** [k9mail/k-9#1189]
### Expected behaviour
I am sending out emails to customers with videos inline. And with an image in it as fallback. Unfortunately the latest K-9 Mail app does not display these.

Here is the HTML I use in these emails (in Jade syntax)
https://github.com/binarykitchen/videomail.io/blob/staging/src/views/email/youhavevideomail.jade

### Actual behaviour
When I load that email in the app, I expect the video to be shown. Instead, it is just black, an empty space wasting lots of pixels.

### Steps to reproduce
1. Send yourself that email. It's my videomail service, see www.videomail.io (record something and send it to you)
2. Open K-9 on your android
3. Load that email and you see, just empty space

### Environment
K-9 Mail version: 5.008

Android version: 4.2.2

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

**Comment by cketti** [comment]
By default K-9 Mail blocks loading of external content. We have some basic detection for `img` tags and display a "Show pictures" button when an `img` element is found pointing to external content. Pressing the button allows remote content to be fetched.

I'm guessing the current mechanism doesn't work with your emails.

I have to say I'm not too thrilled about the idea of supporting videos in emails. Currently I'm leaning towards resolving this as WONTFIX.

**Comment by binarykitchen** [comment]
@cketti understand. but let me explain from a different perspective: this videomail service i have developed is aimed at deaf people like me who use sign language as our first language.

and with videomails, sign language is embedded in emails. this to make us equal to those who read in english.

that's why email apps should become more accessible i think and display at least the inline image and play the video on tap.

hope you understand this from our perspective?

**Comment by cketti** [comment]
I'm guessing the video itself isn't sent via email but hosted on your server. Wouldn't it be easier to attach a video thumbnail and use a link to launch the website when the image is clicked? That should be supported by every email client out there.
Nowadays browsers can't afford not to be able to play videos. So there shouldn't be any problems. Email clients, on the other hand, very often have privacy features that make it hard or impossible to access remote content.

**Comment by binarykitchen** [comment]
@cketti that's right, the video of the videomail itself is hosted on my server.

my problem is that, when sending out the email, i cannot know what the mail agent is on the recipient side. if it is thunderbird, then it fetches the video and plays it inside thunderbird. if it is k-9, it is just an empty, wasted space, making the videomail look ugly and scares away some users of mine.

two questions: 

1. if thunderbird already does it, why not on k-9 too? thunderbird only asks the question once "are you sure you want to fetch the resource from the remote www.videomail.io server?" - hit yes on that and all is fine.

2. if no, then how can i improve the `<video>` tag so that it won't be shown on k-9 so that no big empty space is rendered?

(i already attach a thumbnail and a link to the site, nevertheless you see a big empty space)

**Comment by philipwhiuk** [comment]
RoundCube webmail doesn't do much different. It just shows a pink box. 

I'd be really surprised if this was just a problem limited to K-9 and RoundCube.

**Comment by binarykitchen** [comment]
ping?

**Comment by cketti** [comment]
1. K-9 Mail is unlikely to ever support playing videos inside the app. Right now it might work after pressing the "show pictures" button, but that's just a side-effect of using the system WebView. Future versions of the app might strip unsupported/unwanted elements like `<video>` from messages. I have no intention of supporting all the features of a modern web browser.

2. Also, we have far more important things to work on than optimizing the display of messages that reference external content which is blocked by default. My advice is to avoid this altogether.

Using something like the following should do the trick:
```html
<a href="https://yourvideosite/video_message?id=123">
  <img src="cid:attached_thumbnail_image_cid" alt="Click here to watch the video">
</a>
```

Calling this a video mail might be a stretch. But like I said, I think having video tags inside emails is a bad idea in the first place.

**Comment by binarykitchen** [comment]
thanks @cketti 

i am already using a similar HTML, this one:

```html
<video id="mail" autoplay="autoplay" width="400" height="300" controls="controls" preload="auto" poster="cid:poster.jpg@videomail.io" style="border-radius: 6px;">
    <source src="https://videomail.io/videomail/xyz/type/mp4/" type="video/mp4">
    <source src="https://videomail.io/videomail/xyz/type/webm/" type="video/webm">
    <a href="https://videomail.io/videomail/xyz" style="color: #ea4b2a; outline: 0 none; text-decoration: none;">
        <img src="cid:poster.jpg@videomail.io" width="400" height="300" style="border-radius: 6px;">
    </a>
</video>
```

but k9 does not render this at all. maybe because it is in the video tag? cant you at least code so that it just ignores the video tag and renders the image inside?

and to your last comment, "I think having video tags inside emails is a bad idea in the first place.", i disagree. because we deaf people communicate in sign language and need videos in emails. this is about accessibility!

**Comment by cketti** [comment]
Instead of using the browser which is good at playing video, you're trying to do it inside email clients which traditionally don't support email. You already use the browser to record the video, why not also use it to play the video? The user experience doesn't change dramatically. Recipients only have to click a link/image to get to the browser, an app far better suited for playing videos.
By including a `video` tag you're straying from the well-traveled path of email and bring this kind of incompatibility on yourself.

If you find an easy way to ignore the `video` element and display the link instead I'm happy to merge a pull request.

**Comment by binarykitchen** [comment]
@cketti i understand you but i disagree. www.videomail.io is designed for adding sign language in emails. it sends out the videos inside emails. many email clients already play videos, i.E. Gmail (even its app), Outlook (even its app) and Thunderbird. so tell me why not K9mail too? it is about accessibility. user experience in k9 is worse, videomail users would see an empty white space where the video should be and have to click on the link below which opens the browser. that's a no go for them.

**Comment by binarykitchen** [comment]
ping?
